### DISCUSSION TITLE
I need to pass a variable in a helper inside the innerHTML, help please

### DISCUSSION THREAD

**Discussion by ArturMota19** [handlebars-lang/handlebars.js#1919] [Q&A]
So I have this code and I need to pass the ${nameMatch} to be compared in this helper
![image](https://user-images.githubusercontent.com/108702636/218600100-08b02351-5d9a-4df3-af60-6efb30f0c253.png)
However, when I pass it with quotes, it literally sends "${nameMatch}", and not the value of the nameMatch variable.
And if I remove the quotes, it says I'm trying to pass an invalid type to the helper.
How could I resolve this? Thanks in advance

**Comment by jaylinski** [comment]
I'm not sure what you are trying to do. At what step do you compile the template with Handlebars?

Can you give me a complete example, including the `ifEqual`-helper?

**Reply by ArturMota19** [reply]
I'm new to handlebars, but basically I'm trying to do this dynamically, because for each "match" I need to make that comparison, to see if the name that was passed in the function's parameter is the same that comes from the back-end, understand? The problem is that handlebars does not accept passing variables in the innerHTML with ${example}
-
Function in which I pass the parameter:
![image](https://user-images.githubusercontent.com/108702636/219372923-b8f696e5-5157-4f3d-8ae0-3e8eea74a141.png)
-
Helper if Equal (putting console log in it, it returns me a = ${nameMatch} literally and b=(name that comes from the backend):
![image](https://user-images.githubusercontent.com/108702636/219373330-47ae454c-4413-4750-ba5a-da4ea3f227a8.png)
-
it literally returns the name ${nameMatch} because I put the quotes, because the handlebars does not accept passing a variable in the innerHTML without the quotes. error: Expecting 'CLOSE_RAW_BLOCK', 'CLOSE', 'CLOSE_UNESCAPED', 'OPEN_SEXPR', 'CLOSE_SEXPR', 'ID', 'OPEN_BLOCK_PARAMS', 'STRING', 'NUMBER', 'BOOLEAN', 'UNDEFINED', 'NULL' , 'DATA', 'SEP', got 'INVALID'

**Reply by satyajitnayk** [reply]
@ArturMota19  `ifEqual` helper is just working fine. https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23each%20items%7D%7D%5Cn%7B%7B%23ifEqual%20%40index%200%7D%7D%5Cn%3Ch1%3EHello%3C%2Fh1%3E%5Cn%7B%7Belse%7D%7D%5Cn%3Ch1%3EWorld%3E%3Ch1%3E%5Cn%7B%7B%2FifEqual%7D%7D%5Cn%7B%7B%2Feach%7D%7D%5Cn%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20items%3A%20%5B1%2C2%2C3%2C4%2C5%5D%5Cn%7D%5Cn%22%2C%22output%22%3A%22%3Ch1%3EHello%3C%2Fh1%3E%5Cn%3Ch1%3EWorld%3E%3Ch1%3E%5Cn%3Ch1%3EWorld%3E%3Ch1%3E%5Cn%3Ch1%3EWorld%3E%3Ch1%3E%5Cn%3Ch1%3EWorld%3E%3Ch1%3E%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper('ifEqual'%2C%20function(a%2C%20b%2C%20options)%20%7B%5Cn%20%20%20%20return%20(a%20%3D%3D%20b)%20%3F%20options.fn(this)%20%3A%20options.inverse(this)%3B%20%5Cn%7D)%3B%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D

The error is most probably because the syntax is wrong some where. Fixing it will solve your issue.

<img width="1469" alt="Screenshot 2023-05-16 at 5 37 17 PM" src="https://github.com/handlebars-lang/handlebars.js/assets/32722867/68662ae7-3536-4d45-bb69-f07db67d2d45">
